tcms/static/js/utils.js

```diff
@@ -1,17 +1,16 @@
 /*
     Used to update a select when something else changes.
 */
-function updateSelect(data, selector, id_attr, value_attr) {
+function updateSelect(data, selector, id_attr, value_attr, group_attr) {
     var _select_tag = $(selector)[0];
     var new_options = '';
 
-    // in some cases, e.g. TestRun search, the 1st <option> element is ---
-    // which must always be there to indicate nothing selected
-    if (_select_tag.options.length) {
-        new_options = _select_tag.options[0].outerHTML;
-    }
-
+	currentGroup = '';
     data.forEach(function(element) {
+		if (currentGroup !== element[group_attr] && group_attr != null) {
+			new_options += '<optgroup label="' + element[group_attr] + '">';
+			currentGroup = element[group_attr];
+		}
```

Comment:
This is missing the closing `optgroup` tag and leads to invalid HTML!

Reply:
Fixed